### ISSUE TITLE
Large language models are not zero-shot communicators

### ISSUE THREAD

**Issue by bigshanedogg** [bigshanedogg/survey#27]
## Problem statement

1. LLM (Large Language Model)의 implicature 이해 능력(context 이해) 분석
    - implicature를 이해는 context 이해를 수반하는 어려운 태스크이다. (human 대비 zs 24%, fs 9% 차이)
      - Q: Have you senn my phone?
      - Wrong: Yes, I have seen your phone.
      - Correct: Yes, It is under the table.
    - scale, in-context examples의 수, prompt 구성 방법, context-light vs. context-heavy 별로 성능을 비교
2. LLM의 implicature 이해 능력을 측정할 수 있는 방법론 제시

## Glossary
- implicature: utterances that convey something other than literal meaning

## Baseline
- Cohere
- InstructGPT: 
  - prompt engineering이 중요하다 -> prompt engineering (user intention)에 잘 반응할 수 있도록 학습해보자
  - approach:
    - A) GPT-3를 (prompt, answer)로 fine-tuning
    - B) 33K prompts에 대한 각 4-9개의 answer를 선호도(user intention)에 맞게 레이블링 후 선호도를 예측하는 Reward Model 학습
    - C) A)의 모델을 B)의 Reward Model을 보상함수로 사용하여 강화학습
  - 모델에 explicit constraints (e.g. 답변을 두 문장 이내로 작성하시오)가 주어진 상황에서 더 좋은 성능을 보임

## Data details
- 자체 데이터셋 구축 (https://huggingface.co/datasets/UCL-DARK/ludwig)

## Experiment Setting
- implicature QA에 대해 yes / no 중 1개를 추론하도록 함
- prompt sensitivity
  - natural prompt e.g. "Esther asked 'Can you come to my party on Friday?' and Juan responded 'I have to work', which means no"
  - structured prompt e.g. "Question: Have you found him yet response: They're still looking meaning: no" 
- effect of k when Few-shot in-context evaluation
  <img width="530" alt="스크린샷 2023-01-16 오후 1 48 49" src="https://user-images.githubusercontent.com/18084680/212600883-660c14be-c385-44af-904f-0dd81096190d.png">
- Context-light vs. Context-heavy (implicature의 종류)
  <img width="561" alt="스크린샷 2023-01-16 오후 1 50 47" src="https://user-images.githubusercontent.com/18084680/212601110-c76b3982-beab-40a2-b00d-42bb72adb812.png">
  - Generalised: 쉬움 (context-light), Particularised: 어려움 (context-heavy)

## Evaluation
- Zero-shot evaluation
  - <img width="566" alt="스크린샷 2023-01-16 오후 1 58 10" src="https://user-images.githubusercontent.com/18084680/212601874-5399eedb-cfd6-4464-b8fc-706ac1f512fb.png">
  - natural vs. structured prompt의 차이를 보았을 때, prompt 구성 방식이 모델 간 경향성에 영향을 주지는 않는 것으로 보임
    - <img width="230" alt="스크린샷 2023-01-16 오후 2 32 38" src="https://user-images.githubusercontent.com/18084680/212605398-531acbba-ed0d-428f-af85-aa377537e0c5.png">
    - 기존 prompt engineering 모델 모델 내에서 prompt로 인한 성능 개선을 노려볼 순 있겠으나, 이것이 모델의 본질적인 behavior에 영향을 주진 않는다.
    - InstructGPT는 natural과 structured 사이의 차이가 크지 않다.
  - fig 2 Left를 보면, 가장 큰 모델이 가장 높은 성능을 보이는 모델은 InstructGPT, Cohere, T0 3개이다.
    - <img width="567" alt="스크린샷 2023-01-16 오후 2 10 14" src="https://user-images.githubusercontent.com/18084680/212603082-3e935942-5ac9-48a1-ae48-9a23cc928f17.png">
    - 3개를 제외한 다른 모델은 큰 모델이 작은 모델보다 성능이 나쁜 경우가 있다.
    - InstructGPT를 제외한 다른 모델은 scaling을 해도 성능 개선에 한계가 있는 반면 InstructGPT는 scaling에 따라 유의미한 향상을 보인다.
  - 기존의 implicature 평가는 비교적 쉬운 context-light implicature로 인해 과대평가된 경향이 있다.
    - <img width="556" alt="스크린샷 2023-01-16 오후 2 39 05" src="https://user-images.githubusercontent.com/18084680/212606063-1a276df4-d839-4731-ad7b-5a5d58c7a22f.png">
    - 모델 스케일/아키텍처에 관계 없이 context 이해가 많이 요구된 particularised보다 context 이해가 크게 필요하지 않은 generalized에서 더 높은 성능을 보인다.
      - Cohere-52B: avg 58.5% (gen: 73.9%, par: 51.5%)
      - InstructGPT-3-175B avg 72.3% (gen: 79.3%, par: 59.7%)
    - LLM의 scaling으로 인한 성능 향상은 context resolution이 필요하지 않은 쉬운 examples를 많이 학습한 것 때문으로 보인다.
  - In-context examples에서 k가 성능 향상에 유의미하게 영향을 주는데는 한계가 있었다. (대략 k=5 전후)
    - <img width="546" alt="스크린샷 2023-01-16 오후 2 38 03" src="https://user-images.githubusercontent.com/18084680/212605971-08f76e2f-c525-4065-aa19-c36c60b6a071.png">
    - k가 prompt sensitivity를 낮춰주는 효과는 있었다.

## Conclusion
- 아키테척 별 큰 차이는 없고, multitask FT와 OpenAI의 InstructGPT가 유의미하게 높은 성능을 보임 (여전히 human avg.보다 현저하게 낮지만)
    - scaling과 in-context examples의 수를 늘려도 이 경향성은 크게 달라지지 않는다.
    - implicature resolution을 위해선 단순한 task-specific finetuning이 아닌 instruction finetuning이 필요한 것 같다.

## Limitations
- 현재 LLM의 non-binary implicature resolution에 취약할 것으로 예상하고 있으나, 아직 이를 평가할 방법은 제시되지 않았다.

### Reference and Implementation:
- paper: [Large language models are not zero-shot communicators](https://arxiv.org/abs/2210.14986), 10 Sep 2021
